core/src/main/scala/org/apache/spark/scheduler/cluster/YarnSchedulerBackend.scala

```diff
@@ -91,6 +91,7 @@ private[spark] abstract class YarnSchedulerBackend(
       filterParams.foreach { case (k, v) => conf.set(s"spark.$filterName.param.$k", v) }
       scheduler.sc.ui.foreach { ui => JettyUtils.addFilters(ui.getHandlers, conf) }
     }
+    scheduler.sc.executorAllocationManager.foreach(_.reSetNumExecutorsTarget())
```

Comment:
I guess this should go into some other place. you've put it here into `addWebUIFilter`, just temporary?